core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -281,8 +281,9 @@ public void registerHighlighter(String key, Class<? extends Highlighter> clazz)
         highlighters.registerExtension(key, clazz);
     }
 
-    public void registerSuggester(String key, Class<? extends Suggester> suggester) {
+    public void registerSuggester(String key, Class<? extends Suggester> suggester, NamedWriteable<? extends SuggestionBuilder> builder) {
```

Comment:
@areek @cbuescher One thing I request you both to comment on in particular is adding the `NamedWriteable` parameter to the `registerSuggester` method.  I don't know if this will cause any issues with registering custom suggester plugins, but adding this parameter is required so that we can register the associated builder with the custom suggester in the `namedWriteableRegistry`.